core/execution/spot/market.go

```diff
@@ -0,0 +1,2875 @@
+		orderValidity = "valid"
+	}
+
+	if order.PeggedOrder != nil && order.IsFinished() {
+		// remove the pegged order from anywhere
+		m.removePeggedOrder(order)
+	}
+
+	// insert an expiring order if it's either in the book
+	// or in the parked list
+	if order.IsExpireable() && !order.IsFinished() {
+		m.expiringOrders.Insert(order.ID, order.ExpiresAt)
+	}
+
+	return orderConf, orderUpdates, err
+}
+
+// submitValidatedOrder submits a new order.
+func (m *Market) submitValidatedOrder(ctx context.Context, order *types.Order) (*types.OrderConfirmation, []*types.Order, error) {
+	isPegged := order.PeggedOrder != nil
+	if isPegged {
+		order.Status = types.OrderStatusParked
+		order.Reason = types.OrderErrorUnspecified
+
+		if m.as.InAuction() {
+			order.SetIcebergPeaks()
+			// as the order can't trade we don't transfer from the general account to the holding account in this case.
+			m.peggedOrders.Park(order)
+			// If we are in an auction, we don't insert this order into the book
+			// Maybe should return an orderConfirmation with order state PARKED
+			m.broker.Send(events.NewOrderEvent(ctx, order))
+			return &types.OrderConfirmation{Order: order}, nil, nil
+		}
+		err := m.repricePeggedOrder(order)
+		if err != nil {
+			order.SetIcebergPeaks()
+			m.peggedOrders.Park(order)
+			m.broker.Send(events.NewOrderEvent(ctx, order))
+			return &types.OrderConfirmation{Order: order}, nil, nil // nolint
+		}
+	}
+	var trades []*types.Trade
+	// we're not in auction (not opening, not any other auction
+	if !m.as.InAuction() {
+		// first we call the order book to evaluate auction triggers and get the list of trades
+		var err error
+		trades, err = m.checkPriceAndGetTrades(ctx, order)
+		if err != nil {
+			return nil, nil, m.unregisterAndReject(ctx, order, err)
+		}
+		// NB we don't apply fees here because if this is a sell the fees are taken from the quantity that the buyer pays (in quote asset)
+		// so this is deferred to handling confirmations - by this point the aggressor must have sufficient funds to cover for fees so this should
+		// not be an issue
+	}
+
+	// if an auction is ongoing and the order is pegged, park it and return
+	if m.as.InAuction() && isPegged {
+		if isPegged {
+			m.peggedOrders.Park(order)
+		}
+		m.broker.Send(events.NewOrderEvent(ctx, order))
+		return &types.OrderConfirmation{Order: order}, nil, nil
+	}
+
+	order.Status = types.OrderStatusActive
+
+	// Send the aggressive order into matching engine
+	confirmation, err := m.matching.SubmitOrder(order)
+	if err != nil {
+		return nil, nil, m.unregisterAndReject(ctx, order, err)
+	}
+
+	// if the order is not finished and remaining is non zero, we need to transfer the remaining base/quote from the general account
+	// to the holding account for the market/asset. If an auction is on-going we also need to account for potential fees (applicable for buy orders only)
+	if !order.IsFinished() && order.Remaining > 0 {
+		err := m.transferToHoldingAccount(ctx, order)
```

Comment:
Maybe I can't see it, but if we have a pegged order submitted while we're in auction, it gets parked right away, and we won't do any holding account shuffling. But then I can't see any call to `transferToHoldingAccount` when we leave auction and the parked pegged orders are resubmitted?

Reply:
this should be fixed now.